### PR TITLE
Add dotnet-core-sdk for linux

### PR DESCRIPTION
**Pull request by ZimmerA**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

This is another attempt in getting the dotnet-core-sdk on conda-forge, inspired by https://github.com/conda-forge/staged-recipes/pull/10588 and https://github.com/conda-forge/staged-recipes/pull/8166 .

### CODE REVIEW

**Comment by ZimmerA** [comment]
@conda-forge-admin, please ping team

**Review by beckermr — CHANGES_REQUESTED** [review]
Thanks a bunch for contributing! Looking at the old PRs, there appear to be comments there that have not been addressed here. Please take a look and try your best!

cc @isuruf who reviewed last time

**Comment by ZimmerA** [comment]
Thank you for the review!
Are you referring to https://github.com/conda-forge/staged-recipes/pull/10588#issuecomment-587162721 ?
I added the libraries that caused the errors to the missing dso whitelist and it seemed to have worked out. The package is working after all. 
Is this bad practice? What are my alternatives?

**Comment by beckermr** [comment]
The mac osx min version too, but we should wait for other members of @conda-forge/core. I am not familiar with this framework.

**Comment by ZimmerA** [comment]
After I packaged https://github.com/conda-forge/staged-recipes/pull/12877 and https://github.com/conda-forge/staged-recipes/pull/12922 I managed to get rid of all entries in the missing_dso_whitelist. As opposed to  https://github.com/conda-forge/staged-recipes/pull/12910#issuecomment-711192167 every dependency is now coming from conda-forge and the recipe builds on linux without link errors.

Currently, this recipe is only for linux because we need this urgently to get our Dotnet based tools on conda-forge so they can be used within the galaxy ecosystem https://github.com/galaxyproject. Im planning on iterating the windows and macOS builds on the feedstock in the near future. 

Because of the GLIBC version, the build currently requires the cos7 container to run. I left the cos7 job in so you can see it builds. If this is required I can take it out before the merge. Im going to switch to the cos7 container in the feedstock as recommended by @beckermr.

File: recipes/dotnet-core-sdk/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  - {{ compiler('cxx') }}
+  host:
+  - zlib  # [linux]
+  - libcurl  # [linux]
+  - lttng-ust  # [linux]
+  run:
+  - lttng-ust  # [linux]
+  - libcurl  # [linux]
+  - icu  # [linux]
+  - krb5  # [linux]
```
Comment (isuruf) [inline recipes/dotnet-core-sdk/meta.yaml:27]:
```suggestion
  - krb5  # [linux]
  - __glibc >=2.17  # [linux64]
```

**Comment by bgruening** [comment]
@ZimmerA I think you need to revert your changes to the azure pipelines.

**Comment by ZimmerA** [comment]
> @ZimmerA I think you need to revert your changes to the azure pipelines.
Are you sure?
The problem is that it only builds on the cos7 container which can only correctly be activated on the feedstock. Because of this the build now  fails. As I said in my last comment I talked with @beckermr about this.

Im not sure what the right way to handle this is now.

**Comment by beckermr** [comment]
Right. Leave staged recipes as is. But `skip: True  # [linux]` in your meta.yaml. Then when the feedstock is made, follow the docs to add cos7 to the feedstock.

**Comment by ZimmerA** [comment]
> Right. Leave staged recipes as is. But `skip: True # [linux]` in your meta.yaml. Then when the feedstock is made, follow the docs to add cos7 to the feedstock.

Okay, now we should be good to go. Again thanks for the help!

**Comment by isuruf** [comment]
Let's co-ordinate with @dhirschfeld on #12910

**Comment by ZimmerA** [comment]
Closing this in favor of https://github.com/conda-forge/staged-recipes/pull/12910

**Pull request closed without merging**